eclipse/che
# [Test] Estimate time of java project build and run inside Che Code Editor in performance tests

### What does this PR do?
This PR adds `api load tests` scripts that will be used in DevSpaces performance testing. 

### What issues does this PR fix or reference?
Openshift DevSpaces issue - [Estimate time of java project build and run inside Che Code Editor in performance tests](https://issues.redhat.com/browse/CRW-4044).


### How to test this PR?
1. Create test users on cluster using [performance-users](https://main-jenkins-csb-crwqe.apps.ocp-c1.prod.psi.redhat.com/job/Testing/job/Performance/job/performance-users-creation/) job.
2. Api load test starts by command like:
```
./api-load-tests.sh -u user  -p load-user -s lombok -n 11 -c 1 -b https://devspaces.crw-qe.com -o https://api.ocp410.com:6443
    -u base username 
    -p common password 
    -b URL of DevSpaces 
    -o Openshift cluster URL
    -s test suite (by default its php)
    -n number of users ${NC} usernames will be set in format <username>1, <username>2(by default its 1) 
    -c count on how many times one user should run a workspace(by default its 1)
```
For now `php`, `cpp`, `python`, `lombok` and `quarkus` suites are available for testing.

3. Tests results adds to result.txt with format like:
```
Load testing for 11 users x 1 workspaces took 215 seconds
11 tests passed
0 tests failed
Workspace startup time: 
min: 37 second
avr: 44 second
max: 50 second
Project build time:
min: 140 second
avr: 142 second
max: 146 second
```
All logs from test pods saved in reports folder.

### PR Checklist

[As the author of this Pull Request I made sure that:](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#pull-request-template-and-its-checklist)


### Reviewers

Reviewers, please comment how you tested the PR when approving it.
